platform/spire/src/virt.py

```diff
@@ -491,6 +492,21 @@ def auto_node(ops: setup.Operations, tc: TerminationContext, node: configuration
     ops.add_operation("start up node @HOST", lambda: vm.boot_launch(), node)
 
 
+def auto_nodes_parallel(ops: setup.Operations, tc: TerminationContext, nodes: list, install_iso: str):
+    vms = [VirtualMachine(node, tc, install_iso) for node in nodes]
+
+    def boot_install_and_admit_all():
+        with concurrent.futures.ThreadPoolExecutor(len(vms)) as executor:
+            futures = [executor.submit(vm.boot_install_and_admit) for vm in vms]
+            results = [f.result() for f in futures]
+
+    ops.add_operation("install non-supervisor nodes (this may take several minutes)", boot_install_and_admit_all)
+
+    for node in nodes:
+        vm = VirtualMachine(node, tc)
+        ops.add_operation("start up node @HOST", vm.boot_launch, node)
```

Comment:
Can we also parallelize launching the nodes?

Reply:
Does launching the nodes take a nontrivial amount of time? I have the impression it just starts the process and returns, since the nodes would be kept up for verification.

Reply:
Oh, woops, yeah, you're right.